docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+# Data Sharing Guide: CDM JupyterHub
+
+## Overview
+
+The BER DataLake JupyterHub platform provides data sharing capabilities that allow users to share their datasets with other users and groups. 
+
+All data governance functions are **✨ automatically imported** in every notebook via `startup.py` - no manual imports needed! This guide shows you how to use these pre-loaded functions for seamless data sharing and collaboration.
+
+## Key Concepts
+
+### Data Organization
+
+All user data in BER DataLake is organized under personal namespaces:
+
+- **General Data Storage**: `s3a://cdm-lake/users-general-warehouse/{username}/`
+- **SQL Warehouse**: `s3a://cdm-lake/users-sql-warehouse/{username}/`
+
+## Getting Started
+
+### Auto-Imported Functions
+
+All CDM JupyterHub notebooks automatically import these data governance functions at startup (via `startup.py`), so they're ready to use immediately without any imports:
+
+**Auto-Imported Utility Functions:**
+- `check_governance_health()` - Check service status
+- `get_minio_credentials()` - Get your MinIO credentials (sets environment variables)
+- `get_my_sql_warehouse()` - Get your SQL warehouse prefix
+- `get_my_policies()` - Get detailed policy information
+- `get_my_workspace()` - Get comprehensive workspace information
+- `get_path_access_info(path)` - Check who has access to a path
+- `make_table_private(namespace, table_name)` - Remove public access from SQL tables
+- `make_table_public(namespace, table_name)` - Make SQL tables publicly accessible
+- `share_table(namespace, table_name, with_users, with_groups)` - Share SQL tables
+- `unshare_table(namespace, table_name, from_users, from_groups)` - Unshare SQL tables
+
+**Pre-Initialized Client:**
+- `governance` - Pre-initialized `DataGovernanceClient()` instance for advanced operations
+
+**Other Auto-Imported Functions:**
+- `get_spark_session()` - Create Spark sessions with Delta Lake support
+- Plus many other utility functions for data operations
+
+### Quick Start
+
+```python
+# Check your workspace information
+workspace = get_my_workspace()
+print(f"👤 Username: {workspace.username}")
+print(f"🏠 Home paths: {workspace.home_paths}")
+print(f"👥 Groups: {workspace.groups}")
+print(f"📁 Accessible paths: {len(workspace.accessible_paths)}")
+
+# Check service health
+health = check_governance_health()
+print(f"🔍 Service status: {health.status}")
+
+# Get your credentials (automatically sets environment variables)
+credentials = get_minio_credentials()
+print(f"🔐 Credentials loaded for: {credentials.username}")
```

Comment:
Note that this changes any previous password?

Reply:
Or in the section below

Reply:
👍

Reply:
:+1: